### PR TITLE
Remove `SubscriptionFeatureAvailability` from `AppDependencyProvider`

### PR DESCRIPTION
**Pull request by samsymons**
Task/Issue URL: https://app.asana.com/0/414709148257752/1208561301723448/f
Tech Design URL:
CC:

**Description**:

This PR removes `SubscriptionFeatureAvailability` from `AppDependencyProvider`. Now it instead stores the feature availability instance on the app delegate, and passes it down appropriately, such as through the MainViewController.

**Steps to test this PR**:
1. Smoke test subscription features
2. Check that multiple CI runs are green, to ensure no flaky tests

**Orientation Testing**:

* [ ] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 15
* [ ] iOS 16
* [ ] iOS 17

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: DuckDuckGo/NetworkProtectionRootView.swift [hunk 1]
```diff
@@ -28,9 +28,8 @@ struct NetworkProtectionRootView: View {
 
     init() {
         let accountManager = AppDependencyProvider.shared.subscriptionManager.accountManager
-        let subscriptionFeatureAvailability = AppDependencyProvider.shared.subscriptionFeatureAvailability
         let locationListRepository = NetworkProtectionLocationListCompositeRepository(accountManager: accountManager)
-        let usesUnifiedFeedbackForm = accountManager.isUserAuthenticated && subscriptionFeatureAvailability.usesUnifiedFeedbackForm
+        let usesUnifiedFeedbackForm = accountManager.isUserAuthenticated // && subscriptionFeatureAvailability.usesUnifiedFeedbackForm
```
Comment (samsymons) [inline DuckDuckGo/NetworkProtectionRootView.swift]: This is the one open TODO of this PR, the unified feedback form feature flag is now fully rolled out and we shouldn't need to access it any more, but I'll get confirmation of this before removing it - left it commented out for now as a reminder for myself. In the mean time, the rest of the PR can be reviewed.

File: DuckDuckGo/Subscription/Views/SubscriptionSettingsView.swift [hunk 2]
```diff
@@ -31,7 +31,7 @@ struct SubscriptionSettingsView: View {
     }
     @State var configuration: Configuration
     @Environment(\.dismiss) var dismiss
-    @StateObject var viewModel = SubscriptionSettingsViewModel()
```
Comment (miasma13) [inline DuckDuckGo/Subscription/Views/SubscriptionSettingsView.swift]: Can we leave out this rename. This view model is specific to this view so I have the feeling that adding some naming (that especially with two cases of combination for `subscription` and `settings`) can cause confusion.

Comment (samsymons) [reply DuckDuckGo/Subscription/Views/SubscriptionSettingsView.swift]: Yep, I think we can undo it now that this is no longer being injected in, happy to keep it as-is.

Comment (samsymons) [reply DuckDuckGo/Subscription/Views/SubscriptionSettingsView.swift]: Done ✅

**Review by miasma13 — APPROVED** [review]

**Pull request merged**